jsparagus/gen.py

```diff
@@ -1556,6 +1561,10 @@ def shifted_path_to(self, n, right_of):
         assert isinstance(right_of, list) and len(right_of) >= 1
         if n == 0:
             yield right_of
+            if not self.assume_inconsistent:
```

Comment:
This optimization seems too tricky -- consider removing it or try checking that it is correct with an assertion of some kind...